app.js

```diff
@@ -67,17 +67,14 @@ class ProcessQueue {
             }
             if (this.numProc < 0) this.numProc = 0;
             if (this.numProc < this.maxProc) process.nextTick(this.process.bind(this));
-            fs.readFile(tempPath + '/' + clientid, {encoding: 'utf-8'}, (err, data) => {
-                if (err) {
-                    console.error('Could not open file for store upload', err);
-                    return;
-                }
-                // remove the file
-                fs.unlink(tempPath + '/' + clientid, () => {
-                    // we're good...
+            const path = tempPath + '/' + clientid;
+            store.put(clientid, path);
+                .then(() => {
+                    fs.unlink(path);
+                })
+                .then((err) => {
+                    console.error('Error storing', path, err);
                 });
-                store.put(clientid, data);
```

Comment:
we're removing the file (`path`) only in case of success. Should not we either retry or delete the file?

Reply:
the assumption is that if its not possible to open it doesn't exist. The new code tries to unlink regardless of errors now.